plugins/ptp_operator/ptp4lconf/ptp4lConfig.go

```diff
@@ -0,0 +1,233 @@
+		b.WriteString("  name: " + p.Name + "\n")
+		b.WriteString("  portName: " + p.PortName + "\n")
+		b.WriteString("  role: " + p.Role.String() + "\n")
+		b.WriteString("  portID: " + fmt.Sprintf("%d", p.PortID) + "\n")
+	}
+	return b.String()
+}
+
+//ByInterface ...
+func (Ptp4lCfg *PTP4lConfig) ByInterface(iface string) (PTPInterface, error) {
+	for _, p := range Ptp4lCfg.Interfaces {
+		if p != nil && p.Name == iface {
+			return *p, nil
+		}
+	}
+	return PTPInterface{}, fmt.Errorf("interfaces not found for the interface %s", iface)
+}
+
+//ByPortID ...
+func (Ptp4lCfg *PTP4lConfig) ByPortID(id int) (PTPInterface, error) {
+	for _, p := range Ptp4lCfg.Interfaces {
+		if p != nil && p.PortID == id {
+			return *p, nil
+		}
+	}
+	return PTPInterface{}, fmt.Errorf("interface was not found for the port id %d", id)
+}
+
+// Watcher monitors a file for changes
+type Watcher struct {
+	fsWatcher *fsnotify.Watcher
+	close     chan struct{}
+}
+
+//Close ...
+func (w *Watcher) Close() {
+	close(w.close)
+}
+
+//NewPtp4lConfigWatcher ...
+func NewPtp4lConfigWatcher(dirToWatch string, updatedConfig chan<- *PtpConfigUpdate) (w *Watcher, err error) {
+
+	fsWatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Errorf("error setting for file change notifier %s", err)
+	}
+
+	w = &Watcher{
+		fsWatcher: fsWatcher,
+		close:     make(chan struct{}),
+	}
+
+	go func() {
+		defer w.fsWatcher.Close()
+		//initialize all
+		for _, ptpConfig := range readAllConfig(dirToWatch) {
+			log.Infof("sending ptpconfig changes for %s", *ptpConfig.Name)
+			updatedConfig <- ptpConfig
+		}
+		for {
+			select {
+			case event, ok := <-w.fsWatcher.Events:
+				if !ok {
+					log.Error("could not read watcher events")
+					continue
+				}
+				log.Infof("event: %v", event)
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					if checkIfPtP4lConf(event.Name) {
+						if ptpConfig, err := readConfig(event.Name); err == nil {
+							log.Infof("sending ptpconfig changes for %s", *ptpConfig.Name)
+							updatedConfig <- ptpConfig
+						}
+					}
+				} else if event.Op&fsnotify.Remove == fsnotify.Remove {
+					if checkIfPtP4lConf(event.Name) {
+						fName := filename(event.Name)
+						ptpConfig := &PtpConfigUpdate{
+							Name:      &fName,
+							Ptp4lConf: nil,
+							Removed:   true,
+						}
+						log.Println("config removed file:", event.Name)
+						updatedConfig <- ptpConfig
+					}
+				}
+			case err, ok := <-w.fsWatcher.Errors:
+				if !ok {
+					log.Errorf("cannot read watcher error %s", err)
+					continue
+				}
+				log.Errorf("error while watching for file changes %s", err)
+			}
+		}
+	}()
+	err = w.fsWatcher.Add(dirToWatch)
+	return w, err
+
+}
+
+func readAllConfig(dir string) []*PtpConfigUpdate {
+	var ptpConfigs []*PtpConfigUpdate
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		log.Errorf("error reading all config fils %s", err)
+	}
+	for _, file := range files {
+		if checkIfPtP4lConf(file.Name()) {
+			log.Info(file.Name())
```

Comment:
Is this log temporary? Printing file name without any context is hard to understand in the logs.

Reply:
changed to  log.Infof("ptp4l config file %s",file.Name())

Reply:
I just cleaned it .not required